src/cpp/src/llm_pipeline.cpp

```diff
@@ -77,36 +42,32 @@ EncodedResults beam_search(
     GenerationConfig config
 );
 
-
-class LLMPipeline::LLMPipelineImpl {
+class StatefulLLMPipeline final : public LLMPipelineImplBase {
```

Comment:
I was always curios, why would someone prohibit inheritance by adding `final`. I'm OK with that, but can you explain the reasoning? Especially for a class which isn't declared in a header.